core/lib/contracts/src/consensus_l2_contracts.rs

```diff
@@ -0,0 +1,208 @@
+    
+    pub fn load() -> Self {
+        Self(crate::load_contract(Self::FILE))
+    }
+
+    pub fn bytecode() -> Vec<u8> {
+        crate::read_bytecode(Self::FILE)
+    }
+
+    pub fn get_attester_committee(&self) -> Function<GetAttesterCommittee> { (&self.0).into() }
+
+    pub fn add(&self) -> Function<Add> { (&self.0).into() }
+}
+
+#[derive(Debug, Default)]
+pub struct Secp256k1PublicKey {
+    pub tag: [u8;1],
+    pub x: [u8;32],
+}
+
+fn into_fixed_bytes<const N: usize>(t: Token) -> anyhow::Result<[u8;N]> {
+    match t {
+        Token::FixedBytes(b) => b.try_into().ok().context("bad size"),
+        bad => anyhow::bail!("want fixed_bytes, got {bad:?}"),
+    }
+}
+
+fn into_tuple<const N: usize>(t: Token) -> anyhow::Result<[Token;N]> {
+    match t {
+        Token::Tuple(ts) => ts.try_into().ok().context("bad size"),
+        bad => anyhow::bail!("want tuple, got {bad:?}"),
+    }
+}
+
+fn into_uint<I:TryFrom<ethabi::Uint>>(t: Token) -> anyhow::Result<I> {
+    match t {
+        Token::Uint(i) => i.try_into().ok().context("overflow"),
+        bad => anyhow::bail!("want uint, got {bad:?}"),
+    }
+}
+
+impl Secp256k1PublicKey {
+    fn from_token(token: Token) -> anyhow::Result<Self> {
+        let [tag, x] = into_tuple(token)?;
+        Ok(Self {
+            tag: into_fixed_bytes(tag).context("tag")?,
+            x: into_fixed_bytes(x).context("x")?,
+        })
+    }
+
+    fn into_token(self) -> Token {
+        Token::Tuple(vec![
+            Token::FixedBytes(self.tag.into()),
+            Token::FixedBytes(self.x.into()),
+        ])
+    }
+}
+
+#[derive(Debug)]
+pub struct Attester {
+    pub weight: u32,
+    pub pub_key: Secp256k1PublicKey,
+}
+
+impl Attester {
+    fn from_token(token: Token) -> anyhow::Result<Self> {
+        let [weight, pub_key] = into_tuple(token)?;
+        Ok(Self {
+            weight: into_uint(weight).context("weight")?,
+            pub_key: Secp256k1PublicKey::from_token(pub_key).context("pub_key")?,
+        })
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BLS12_381PublicKey {
+    a: [u8;32],
+    b: [u8;32],
+    c: [u8;32],
+}
+
+impl BLS12_381PublicKey {
+    fn into_token(self) -> Token {
+        Token::Tuple(vec![
+            Token::FixedBytes(self.a.into()),
+            Token::FixedBytes(self.b.into()),
+            Token::FixedBytes(self.c.into()),
+        ])
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BLS12_381Signature {
+    a: [u8;32],
+    b: [u8;16],
+}
+
+impl BLS12_381Signature {
+    fn into_token(self) -> Token {
+        Token::Tuple(vec![
+            Token::FixedBytes(self.a.into()),
+            Token::FixedBytes(self.b.into()),
+        ])
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct AddInputs {
+    node_owner: ethabi::Address,
+    validator_weight: u32,
+    validator_pub_key: BLS12_381PublicKey,
+    validator_pop: BLS12_381Signature,
+    attester_weight: u32,
+    attester_pub_key: Secp256k1PublicKey,
+}
```

Comment:
I understand it's draft but just in case - please make the public interface and intended use of this file a little more explicit. Maybe just adding doc comments and rearranging the order of structs here will be sufficient